src/main/java/uk/gov/hmcts/ccd/domain/model/aggregated/CompoundFieldOrderService.java

```diff
@@ -0,0 +1,146 @@
+package uk.gov.hmcts.ccd.domain.model.aggregated;
+
+import com.google.common.collect.Lists;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseEventFieldComplex;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseField;
+import uk.gov.hmcts.ccd.domain.model.definition.FixedListItem;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.Integer.valueOf;
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.comparingInt;
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.substringAfterLast;
+
+/**
+ * This service sorts nested fields of a compound field (complex or collection) in the following order:
+ * 1) If caseEventComplexFields is non empty array it will sort on its elements order values.
+ *      (EventToComplexTypes has at least one value for DisplayOrder column defined it will sort according to this column)
+ * 2) If caseEventComplexFields is empty array then it will sort caseFields' children order values.
+ *      (ComplexTypes tab's DisplayOrder column)
+ *
+ * Additionally it sorts the field type's FixedListItem list of case field of FixedList, FixedRadioList and MultiSelectList types.
+ */
+@Named
+@Singleton
+public class CompoundFieldOrderService {
```

Comment:
I will try to move it to definition store now.

Reply:
Done now - the compound case field complex fields ordering as well as fixed list items ordering are done in def store now.